packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -459,6 +463,10 @@ const DataGrid = injectIntl((props) => {
       event.preventDefault(); // prevent the page from moving with the arrow keys.
       return;
     }
+
+    const newRowId = nextRow === 0 ? undefined : dataGridRows[nextRow - 1]?.id;
```

Comment:
I don't think this logic will work with sections.  You might have to get the row identifier from the table row element.